common/src/main/java/com/microsoft/identity/common/internal/fido/WebAuthnJsonUtil.kt

```diff
@@ -0,0 +1,78 @@
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.fido
+
+import com.microsoft.identity.common.java.logging.Logger
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * A utility class to help convert to and from strings in WebAuthn json format.
+ */
+class WebAuthnJsonUtil {
+    companion object {
+        private val TAG = WebAuthnJsonUtil::class.simpleName
+        /**
+         * Takes applicable parameters from an AuthFidoChallenge object and creates a string
+         * representation of PublicKeyCredentialRequestOptionsJSON (https://w3c.github.io/webauthn/#dictdef-publickeycredentialrequestoptionsjson)
+         * @param challengeObject AuthFidoChallenge
+         * @return a string representation of PublicKeyCredentialRequestOptionsJSON.
+         */
+        @JvmStatic
+        fun createJsonAuthRequestFromChallengeObject(challengeObject: AuthFidoChallenge): String {
+            val methodTag = "$TAG:CreateJsonAuthRequestFromChallengeObject"
+            //Create classes
+            val publicKeyCredDescriptorList = ArrayList<PublicKeyCredentialDescriptor>()
+            challengeObject.allowedCredentials?.let { allowedCredentials ->
+                for (id in allowedCredentials) {
+                    publicKeyCredDescriptorList.add(
+                        PublicKeyCredentialDescriptor(
+                            type = "public-key",
+                            id = id
+                        )
+                    )
+                }
+            }
+
+            val request = PublicKeyCredentialRequestOptions(
+                challenge = challengeObject.challenge,
+                rpId = challengeObject.relyingPartyIdentifier,
+                allowCredentials = publicKeyCredDescriptorList,
+                userVerification = challengeObject.userVerificationPolicy
+            )
+
+            try {
+                return Json.encodeToString(request)
+            } catch (e: Exception) {
+                //This would be either SerializationException or IllegalArgumentException,
+                // and both are very unlikely to be thrown since we're only working with one class,
+                // but this catch block is here in case I'm wrong about that, or somethings changes
+                // in the future.
+                val errorMessage = "Error while encoding PublicKeyCredentialRequestOptions to string."
+                Logger.error(methodTag, errorMessage, e)
+                throw Exception(errorMessage + " " + e::class.simpleName + ": " + e.message)
```

Comment:
Can we just not throw the original exception back?

Reply:
Looking at this part again, I think I will change this to throw the original exception back, at the very least for the reason that it's confusing without the rest of the PRs out yet. 
(Long answer for a short question, to explain thinking)
My current plan is to handle the exceptions, telemetry, and final result to the PasskeyChallengeHandler (not out for review yet), but I wanted to avoid the challenge handler from "knowing too much". Particularly that applies to CredMan, which is going to be housed in a CredManManager object and can throw various dependency-specific exceptions. I don't want the challenge handler to directly catch certain Exception types, but many of these exceptions don't have messages (not sure if that's because CredMan is in beta?), and the challenge handler wouldn't know what the exception was for without checking the exception's type. 
So, I was thinking that for dependency specific exceptions, I would catch them in their manager/util class, wrap them in a generic Exception with an actual message written by us, and throw that exception. That Exception would be caught by challenge handler, which will record telemetry including the message and then send that message to ESTS as an error response to the protocol.
I think a lot of this makes little sense without the PRs to demonstrate what I'm thinking :) so will remove the catch block of this method for now.

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2172/commits/1916a14746ab487e73be1feb62b47af64d40ee7c